enhancements/platforms/aws/customer-provisioned-vpc-and-subnets.md

```diff
@@ -0,0 +1,240 @@
+- Control plane TCP 6443 ingress (Kubernetes API).  We can restrict this to the API load balancers.  Folks hitting localhost will not get out to the security group level, and everyone else should be coming in via the load balancers.
+- Control plane TCP 22623 ingress (MCS): this should be restricted to the control plane and compute security groups, instead of the current by-VPC-CIDR logic to avoid leaking sensitive Ignition configs to non-cluster entities sharing the VPC.
+
+
+### Risks and Mitigations
+
+#### Tagging subnets
+
+Currently Terraform does not support management of resources that already exist, so we cannot tag the subnets using Terraform’s aws_subnet resource or data_sources.  Instead, we will tag the subnets from Go immediately before applying the Terraform configuration.
+
+#### Destroy
+
+Destroy will remain mostly unchanged, hinging on the kubernetes.io/cluster/.\*: owned tag.  We will need to grow new code to store kubernetes.io/cluster/.\*: shared in metadata.json and remove it from resources on which it is found.
+
+We should probably revert:
+- #1268, which began removing instance profiles by name.  That was a workaround to recover from openshift-dev clusters which were partially-deleted by the DPP reaper.  Folks using the installer’s destroy code won’t need it, and while the risk of accidental name collision is low, I don’t think it’s worth taking that risk.
+- #1704, which skipped by-tag network interface deletion to reduce AWS API load.  But since #2169, we will be able to have by-tag interface deletion without a throttle-inducing load.
+
+## Design Details
+
+### Test Plan
+
+CI jobs should set up a VPC, subnets, and routing using our UPI flow and launch the cluster-under-test into that infrastructure.  We may want to adjust the resource grouping in our CloudFormation templates to do this, e.g. shifting the S3 endpoint to a later or separate template to show that users don’t need to provide it.  We will not add CI running multiple clusters inside a single VPC at the moment, although we may come back and add this later.
+
+### Graduation Criteria
+
+This enhancement will follow standard graduation criteria.
+
+##### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+
+##### Tech Preview -> GA
+
+- Community Testing
+- Sufficient time for feedback
+- Upgrade testing from 4.3 clusters utilizing this enhancement to later releases
+- Downgrade and scale testing are not relevant to this enhancement
+
+**For non-optional features moving to GA, the graduation criteria must include
+end to end tests.**
+
+## Drawbacks
+
+TODO
+
+## Alternatives
+
+TODO, some can be ported from this private document https://docs.google.com/document/d/1eNtnrsMUL2efRC5Y8w6CL4f_GxTHrIij8cnyQQnUhW8/edit#heading=h.ttvcmxr8sl3b
```

Comment:
@wking can i get your thoughts on which alternatives items we want to port over from the design document? Most seemed related to CI rather than the implementation itself.